libs/auth-api-lib/src/lib/delegations/delegations-incoming-custom.service.ts

```diff
@@ -0,0 +1,308 @@
+import { Inject, Injectable, Logger } from '@nestjs/common'
+import { ConfigType } from '@nestjs/config'
+import { InjectModel } from '@nestjs/sequelize'
+import { uniqBy } from 'lodash'
+import { ClientAllowedScope } from '../clients/models/client-allowed-scope.model'
+import { ApiScope } from '../resources/models/api-scope.model'
+import { DelegationConfig } from './DelegationConfig'
+import { DelegationDTO } from './dto/delegation.dto'
+import { MergedDelegationDTO } from './dto/merged-delegation.dto'
+import { DelegationScope } from './models/delegation-scope.model'
+import { Delegation } from './models/delegation.model'
+import { DelegationValidity } from './types/delegationValidity'
+import { getScopeValidityWhereClause } from './utils/scopes'
+import * as kennitala from 'kennitala'
+import {
+  IndividualDto,
+  NationalRegistryClientService,
+} from '@island.is/clients/national-registry-v2'
+import { LOGGER_PROVIDER } from '@island.is/logging'
+import { isDefined } from '@island.is/shared/utils'
+import { partitionWithIndex } from './utils/partitionWithIndex'
+import { AuditService } from '@island.is/nest/audit'
+import { DelegationsOutgoingService } from './delegations-outgoing.service'
+import { Features } from '@island.is/feature-flags'
+import { FeatureFlagService } from '@island.is/nest/feature-flags'
+
+export const UNKNOWN_NAME = 'Óþekkt nafn'
+
+/**
+ * Service class for incoming delegations.
+ * This class supports domain based delegations.
+ */
+@Injectable()
+export class DelegationsIncomingCustomService {
+  constructor(
+    @InjectModel(Delegation)
+    private delegationModel: typeof Delegation,
+    @InjectModel(ClientAllowedScope)
+    private clientAllowedScopeModel: typeof ClientAllowedScope,
+    @Inject(DelegationConfig.KEY)
+    private delegationConfig: ConfigType<typeof DelegationConfig>,
+    private nationalRegistryClient: NationalRegistryClientService,
+    private featureFlagService: FeatureFlagService,
+    @Inject(LOGGER_PROVIDER)
+    private logger: Logger,
+    private auditService: AuditService,
+    private delegationsOutgoingService: DelegationsOutgoingService,
+  ) {}
+
+  async findAllValidIncoming(
+    user: User,
+    domainName?: string,
+  ): Promise<DelegationDTO[]> {
+    const feature = await this.featureFlagService.getValue(
+      Features.customDelegations,
+      false,
+      user,
+    )
+    if (!feature) {
+      return []
+    }
+
+    const { delegations, fromNameInfo } = await this.findAllIncoming(
+      user,
+      DelegationValidity.NOW,
+      domainName,
+    )
+
+    const delegationDTOs = delegations.map((d) => d.toDTO())
+
+    return delegationDTOs.map((d) => {
+      const person = this.getPersonByNationalId(fromNameInfo, d.fromNationalId)
+
+      return {
+        ...d,
+        fromName: person?.name ?? d.fromName ?? UNKNOWN_NAME,
+      }
+    })
+  }
+
+  async findAllAvailableIncoming(user: User): Promise<MergedDelegationDTO[]> {
```

Comment:
Maybe this should return DelegationDTOs and let the parent function handle delegation merging instead of doing uniqBy here?

Then this function is exactly the same as above, except it filters delegations/scopes by a client. Could be named or refactored based on that.